src/main/java/com/cavalcantgus/book_recommendation_system/services/RatingService.java

```diff
@@ -0,0 +1,78 @@
+package com.cavalcantgus.book_recommendation_system.services;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.stereotype.Service;
+
+import com.cavalcantgus.book_recommendation_system.entities.Rating;
+import com.cavalcantgus.book_recommendation_system.exceptions.DatabaseException;
+import com.cavalcantgus.book_recommendation_system.exceptions.ResourceNotFoundException;
+import com.cavalcantgus.book_recommendation_system.repositories.RatingRepository;
+
+@Service
+public class RatingService {
+
+	private final static Logger logger = LoggerFactory.getLogger(RatingService.class);
+
+	@Autowired
+	private RatingRepository repository;
```

Comment:
```suggestion
	private final RatingRepository repository;
```

daí anota a classe com `@RequiredArgsConstructor` do `lombok`. Essa é a maneira mais recomendada atualmente. Usar no construtor, não nas propriedades. E fica super simples, pq parece mágica que o bean vai ser injetado (mas por debaixo dos panos, o spring vai usar o construtor criado pelo lombok).

> Tu vai anotar a classe pro lombok criar o construtor pra ti. Mas nada impede que criasse um construtor e anotar esse construtor com `@Autowired`. Mas o caminho é sempre escrever menos código e aproveitar bem as bibliotecas.